### PR TITLE
getblocks retry 5 times before giving up

### PR DESCRIPTION
**Pull request by RCasatta**

### CODE REVIEW

File: src/new_index/fetch.rs [hunk 1]
```diff
@@ -82,9 +82,24 @@ fn bitcoind_fetcher(
...
-                    .getblocks(&blockhashes)
-                    .expect("failed to get blocks from bitcoind");
+
+                let mut attempts = 5;
+                let blocks = loop {
+                    attempts -= 1;
+
+                    match daemon.getblocks(&blockhashes) {
+                        Ok(blocks) => break blocks,
+                        Err(e) => {
```
Comment (shesek) [inline src/new_index/fetch.rs:92]: Perhaps this should explicitly check for "Block not found on disk" errors and terminate the thread for other unexpected errors?

Comment (RCasatta) [reply src/new_index/fetch.rs:92]: On a second thought non recoverable errors will panic anyway after 5 attempts, while recoverable errors will leave a trace in the logs. Are we sure we don't want to retry with any error?

Comment (RCasatta) [reply src/new_index/fetch.rs:92]: Anyway, I opened #118 which retry only on "Block not found on disk", one or the other is fine for me

Comment (shesek) [reply src/new_index/fetch.rs:92]: Thanks. I generally prefer to fail fast, if both are fine by you then I would go with #118

(Perhaps also worth mentioning that the daemon already detects connection errors and retries for all RPC commands, so the explicit "Block not found on disk" handling here is done on top of that behavior.)

**Comment by RCasatta** [comment]
closing in favor of #118

**Pull request closed without merging**